content/src/service/deployments/deployments.ts

```diff
@@ -32,13 +32,19 @@ export async function getDeployments(
 
   const moreData = deploymentsWithExtra.length > curatedLimit
 
-  const deploymentsResult = deploymentsWithExtra.slice(0, curatedLimit)
+  let deploymentsResult = deploymentsWithExtra.slice(0, curatedLimit)
+
   const deploymentIds = deploymentsResult.map(({ deploymentId }) => deploymentId)
+
   const content = await getContentFiles(components, deploymentIds)
 
   // TODO [new-sync]: migrationData nolonger required
   const migrationData = await getMigrationData(components, deploymentIds)
 
+  if (!options?.includeDenylisted) {
+    deploymentsResult = deploymentsResult.filter((result) => !components.denylist.isDenyListed(result.entityId))
+  }
```

Comment:
Do you prefer to run this outside the function?

Reply:
There isn't a way to run this outside the function, because if it's done outside the information will not be there, the only thing we could do is outside the function build an object that has ```{ entityId, isDenyListed: true }```